torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -128,6 +128,15 @@ class ProcessGroup {
       std::vector<at::Tensor>& inputTensors,
       const AllgatherOptions& opts = AllgatherOptions()) = 0;
 
+  // Q: Should we move it to protected or annotate with _ since it can be
+  // confusing to use and is intended for internal use only?
```

Comment:
We could expose it for power users. I think it's fine to leave public. Btw, didn't we call it `single` before?

Reply:
I think we agreed on _base during our meeting with @mrshenli